### PR TITLE
Various small fixes

### PR DESCRIPTION
**Pull request by Antolius**
I've had some issues using Watson locally. These are the fixes I needed to make to the original templates:

1. Fixed Colors.vala file name and added the generated config file to executable sources in meson build. 
2. Added missing string quotes in Application.vala. 
3. Renamed the Config.vala template file into Config.vala.in (which is how meson build refers to it). 
4. Fixed path typo in POTFILES.

I also added temp files created by elementary OS' Code editor to gitignore (both Watson's and in the template). My reasoning here was that since it's the default code editor on the platform it should be OK to support this out of the box.

### CODE REVIEW

File: template/meson.build [hunk 1]
```diff
@@ -31,7 +31,8 @@ executable(
     'src' / 'Application.vala',
     'src' / 'MainWindow.vala',
     'src' / 'Widgets' / 'HeaderBar.vala',
-    'src' / 'Constants' / 'Colours.vala',
+    'src' / 'Constants' / 'Colors.vala',
+    config_header,
```
Comment (Antolius) [inline template/meson.build:35]: The `config_header` here is a reference to the `Config.vala` file created on lines 13-17.

**Comment by Suzie97** [comment]
@Antolius, you might want to change the "Elementary OS 6 (Odin)" to "elementary OS 6 (Odin)" in the readme generated by the template, as according to the brand guidelines, that is how the name should be used.

Screenshot 2021-12-05 at 16 42 13

Its a super small fix, so I don't want to open a new PR for it. And it should go well with this PR.

**Comment by aral** [comment]
@Antolius Thanks so much for this, Josip. I’d just ported a bunch of changes back from Comet into Watson and hadn’t had a chance to test them properly yet. Sounds like you just saved me some work (and apologies for the hassle).

@Suzie97  Thanks for the heads up Rajdeep, I’ll make sure this gets implemented. No need to update the PR.

**Comment by aral** [comment]
@Suzie97 Fixed in 95b1d6c

**Comment by Antolius** [comment]
@aral no problem! It's a cool project, I'm glad I could have helped, even a little.

**Pull request merged**